### ISSUE TITLE
No Battery percentage sensor despite data being available (for Enphase / Remote Meter Kit setup)

### ISSUE THREAD

**Issue by joergbattermann** [SpanPanel/Span#26]
Hi there!

I updated to v1.0.2 yesterday evening after seeing the new release and that @pavandave had provided initial support for Battery percentage (thanks!!) because we do have a Span Panel + its Enphase Remote Meter kit to work with / monitor our Enphase setup (Solar + Batteries), i.e.:

![IMG_3262](https://github.com/user-attachments/assets/0be3c278-d53e-4a3e-85fb-d4ce5b807876)

Now what I hoped would happen after upgrading from v0.0.12 to v1.0.2 was that the described 'Battery Percentage' sensor would just show up.. which it didn't. So I uninstalled the integration, restarted HA, redownloaded the integration in HACS, restarted HA again.. and then (re)added it to HA.

Unfortunately this also didn't add the 'Battery Percentage' sensor (everything else is still/again there of course) & so I went ahead and checked the HA logs for calls/responses per the [const.py](https://github.com/SpanPanel/Span/blob/main/custom_components/span_panel/const.py)'s _storage/soe_ endpoint and yep.. it's there as per debug logs (HA integration etc re-add/re-load etc took a couple of minutes & hence the state of charge is higher in the data below given the batteries are/were indeed charging up):

```
2024-09-22 13:01:42.341 DEBUG (MainThread) [custom_components.span_panel.span_panel_api] HTTP GET Attempt #1: http://192.168.7.71/api/v1/storage/soe
2024-09-22 13:01:42.365 DEBUG (MainThread) [custom_components.span_panel.span_panel_api] Fetched from http://192.168.7.71/api/v1/storage/soe: <Response [200 OK]>: {"soe":{"percentage":74}}
2024-09-22 13:01:42.365 DEBUG (MainThread) [custom_components.span_panel.coordinator] Finished fetching span panel SpanPanel data in 0.841 seconds (success: True)
```

There's no disabled sensor to enable but maybe I'm just doing / expecting it wrong.. should the 'Battery Percentage' sensor simply show up or would I have to do something for it to do so... or is there something else maybe a bit different / off or something?

Thanks!

**Comment by pavandave** [comment]
Can you disable and reenable the sensor in the config flow?

**Comment by cayossarian** [comment]
You need to enable the sensor in the Span Panel Options...Settings->Devices->Span Panel->Configure

![span configure](https://github.com/user-attachments/assets/043e026e-d9be-400e-b32a-8e2a8a515b8b)

**Comment by joergbattermann** [comment]
Ha!! I hadn't looked & therefore seen that configuration option, my bad & thank you @cayossarian / @pavandave for the super-fast reply, _much_ appreciated!

And yes, it does show up now:

![image](https://github.com/user-attachments/assets/8d79c173-eb92-454a-b774-6c54254ce9fa)

The only small feedback I have would be the naming of that sensor.. as it isn't fully in line (I think?) with the rest of the system entities.. the other ones (i.e. 'Current Power', 'Cellular Link', 'Door State', 'Software Version' etc) aren't prefixed with 'SPAN*' in their name.


Oh and a small side topic: the Span solar integration for the Enphase systems is a bit different (see the [aforementioned Remote Meter Kit](https://techportal.span.io/s/remote-meter-kit)) as it doesn't have the solar breakers inside the Span panel (or not necessarily so), but rather in the Enphase components (in the Combiner as far as I remember). The HA configuration seems to require a non-zero leg value for l1 / l2 which.. doesn't apply here. Is there an easy way to find out if or what of the Span panel's APIs returns the solar data nevertheless / maybe differently than the Span-integrated-solar-circuits way?

Either way, thanks again Pavan & Bill!

**Comment by cayossarian** [comment]
Thanks for the feedback on the battery sensor name.  Of course you can rename the sensor as a work around for now.

You may find the feedback of interest from a SPAN employ in a previously [closed issue](https://github.com/SpanPanel/Span/issues/10) - see comment from mmbush.  Bottom line is the SPAN API does not currently report solar and the solar configuration in this integration is a work around for those that had solar installed on the SPAN legs itself.  

I would gladly expose any API SPAN decided to implement but so far no luck.  Maybe open an issue with SPAN and they may find it worthwhile at some point.  It appears most of their efforts are related to cloud.  Squeaky wheel gets the oil.

**Comment by joergbattermann** [comment]
Ahh right, could have searched for an existing inqury/issue about that but again, thank you!

Not sure if Matt / @mbbush has a minute to chime in if the local panel API exposes the Enphase Remote Meter kit readings or whether it's quasi the same / similar situation as the SolarEdge side of things.

**Comment by mbbush** [comment]
I'm afraid the local API doesn't expose any information about the span remote meter kit. That product was designed and built long after the local API, and unfortunately we never updated the API to add that support. 

I'll ask around internally about what it would take to add it, but don't get your hopes up.

I don't have an enphase battery myself, so I don't know if they have an API you can integrate with.

**Comment by joergbattermann** [comment]
Thanks for taking the time & the info @mbbush .. appreciate it!

As a bit of a generic FYI re: Enphase: they a have an [officially supported cloud based API](https://developer-v4.enphase.com/) for their ecosystem which has, unfortunately, quite low & realistically unusable monthly call quotas and [the prices for the non-free tiers](https://developer-v4.enphase.com/developer-plans) go beyond a typical household budget.

They also have a local API which is running on their local gateway component (which is quasi the piece of hardware that connects your local install to their cloud) which was at one point [somewhat, yet never fully,] documented(https://enphase.com/download/iq-gateway-access-using-local-apis-or-local-ui-token-based-authentication-tech-brief) and, _afaik_, also never officially supported by Enphase. The [HA integration](https://www.home-assistant.io/integrations/enphase_envoy/) and underlying library use exactly that local _Envoy_ API but that one has slowly been falling apart*. For the time being the CTs readings / (dis)charge and grid import/export readings are still available [but local control functionality no longer fully works](https://github.com/home-assistant/core/issues/123216#issuecomment-2299683614).

Anyway, my particular case &  reason why _I_ was interested in the HA SPAN intgeration to provide the remote meter kit's readings is a bit more special. SPAN & its remote meter kit _seemingly_ don't work a 100% correctly with Enphase's latest generation of 'IQ' batteries just yet - I've had a support case open for some time now (if you're interested, the ticket no is 24241) & actually your colleagues made good progress with more recent firmware updates bringing major improvements with readings getting 'closer'.

The state of charge and (dis)charge grid import/export readings that Span sees/reports and what Enphase says is "actually" happening unfortunately still diverge during _some_ circumstances however which therefore still lead to incorrect SoC "assumptions" for the SPAN side of things vs what's actually in the batteries... and therefore potentially incorrect load-shedding during outages.

I obviously want to make life easier for your colleagues but I simply _can't_ keep an eye on both ecosystems' apps etc 24/7 during all possible loads/discharge/charge/import/export conditions. Hence the idea / brief hope that the HA integeration _could_ potentially access/provide the SPAN's remote meter kit data/readings and it would make extracting & comparing the two systems' readings much easier, bring them into better alignment & maybe figure out what's going on & under what circumstances.

**TL;DR:** thanks @mbbush  & _IF_ you find the time to ask somebody internally about that.. great.. and if not or if it's also a negative re: doable.. all good as well, I appreciate it either way!

@cayossarian .. I'll close this particular issue given the side-topic / Enphase <> Span Solar/Battery readings being exposed and potentially usable for the integrationI is quasi unrelated to the actual topic of the issue and may just be a _very_-low-hope-future-feature one which I could/would then potentially contribute myself. Again, @mbbush no hope / no pressure 😉 



*Re: local Enphase / Envoy API "falling apart": It _appears_ as if Enphase is shifting away from the past model of said 'gateway' being the local and generally 'primary' brain that controls, keeps the actual system state and -mode and just keeps the cloud up to date secondarily.. to a model where the cloud is quasi the primary state-keeper and -controller and the gateway just receives whatever changes you make / mode you select in the cloud and merely regularly reports energy readings back.
